qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -90,6 +105,115 @@ def quantum_instance(
         else:
             self._quantum_instance = quantum_instance
 
+    @property
+    def free_parameters(self) -> Union[ParameterVector, Sequence[Parameter]]:
+        """Return the vector of free parameters."""
+        return self._free_parameters
+
+    @free_parameters.setter
+    def free_parameters(self, free_params: Union[ParameterVector, Sequence[Parameter]]):
+        """Sets the free parameters"""
+        if free_params:
+            self._free_param_binds = {free_params[i]: free_params[i] for i, _ in enumerate(free_params)}
+        else:
+            self._free_param_binds = None
+
+        self._free_parameters = free_params
+
+    def assign_free_parameters(
+        self, values: Union[Mapping[Parameter, ParameterValueType],
+                            Sequence[ParameterValueType]]
+    ) -> None:
+        """
+        Assign free parameters in the QuantumKernel feature map.
+
+        Args:
+            values (dict or iterable):
+                Dict {parameter: value, ...} or {parameter: parameter}:
+                The keys of the parameter dictionary must be Parameter instances in the
+                feature map circuit. The values of the dictionary should be either numeric
+                values {param:float} or copies of the param key {param:param}. Passing a copy of the param
+                key as a value causes that parameter to be ignored. This allows the circuit's
+                free parameters to be bound consecutively.
+
+                Iterable [value0, value1, ..., valueN]:
+                If a list of real values is passed, the elements are assigned to the parameters
+                stored in the free_parameters field.
+
+        Return:
+            None
+
+        Raises:
+            ValueError: Incompatible number of free parameters and values
+        """
+        if not isinstance(values, dict):
+            if self._free_parameters is None:
+                raise ValueError(
+                    f"""
+                    An iterable of free parameter values was given as input,
+                    but the number of parameter values ({len(values)}) does not match the
+                    number of free parameters tracked by the QuantumKernel (None).
```

Comment:
Wouldn't it be simpler for the error message to be more like 'An iterable of free parameters was given but no free parameters were defined for the QuantumKernel` - it matched before more the length mismatch one below but I think its kinda wordy/complicated for this case